client-js/app/elements/labrad-plot.ts

```diff
@@ -303,26 +451,111 @@ export class Plot extends polymer.Base {
       const gradientFill = `url('${location.href}#ColorBarGradient')`;
 
       // Color Bar Rectangle
-      const colorBarOffset = width + COLOR_BAR_LEFT_MARGIN;
       p.svg.append('rect')
+             .attr('id', 'color-bar')
              .attr('fill', gradientFill)
-             .attr('transform', `translate(${colorBarOffset}, 0)`)
              .attr('width', COLOR_BAR_WIDTH)
-             .attr('height', height)
              .attr('stroke', '#000000')
              .attr('stroke-width', COLOR_BAR_STROKE_WIDTH);
 
       // Z-axis ticks and label.
-      const zAxisOffset = width + COLOR_BAR_LEFT_MARGIN + COLOR_BAR_WIDTH + COLOR_BAR_RIGHT_MARGIN;
       p.svg.append('g')
-             .attr('class', 'z axis')
-             .attr('transform', `translate(${zAxisOffset}, 0)`)
-             .call(p.zAxis);
+             .attr('id', 'color-bar-ticks')
+             .attr('class', 'z axis');
+    }
+
+    p.updatePlotStyles();
+    p.updateControlEventListeners();
+  }
+
+
+  /**
+   * Resizes the WebGL renderer and camera.
+   */
+  private resizeWebGLPlot() {
+    this.camera.aspect = this.width / this.height;
+    this.camera.updateProjectionMatrix();
+    this.renderer.setSize(this.width, this.height);
+    this.$.canvas.style.top = `${this.margin.top}px`;
+    this.$.canvas.style.left = `${this.margin.left}px`;
+  }
+
+
+  /**
+   * Resize the SVG portion of the plot.
+   */
+  private resizeSVGPlot(): void {
+    const plot = this.$.plot;
+    const plotBounds = plot.getBoundingClientRect();
+    const plotWidth = Math.max(plotBounds.width, 400);
+    const plotHeight = Math.max(plotBounds.height, 400);
+
+    // The inner dimensions of the plot, sans margins
+    this.width = plotWidth - this.margin.left - this.margin.right;
+    this.height = plotHeight - this.margin.top - this.margin.bottom;
+
+    this.xScale.range([0, this.width]);
+    this.yScale.range([this.height, 0]);
+    this.zScale.range([this.height, 0]);
+
+    this.xAxis.scale(this.xScale)
+              .tickSize(-this.height);
+
+    this.yAxis.scale(this.yScale)
+              .tickSize(-this.width);
+
+    this.line.x((d) => this.xScale(d[0]))
+             .y((d) => this.yScale(d[1]));
+
+    this.zoom.x(this.xScale)
+             .y(this.yScale);
+
+    this.svg.select('#rect')
+              .attr('width', this.width)
+              .attr('height', this.height);
+
+    this.svg.select('#x-axis')
+              .attr('transform', `translate(0, ${this.height})`)
+              .call(this.xAxis);
+
+    const xLabXOffset = this.width / 2;
+    const xLabYOffset = this.height + this.margin.bottom - 10;
```

Comment:
I was confused about "Lab" here, thought it meant "lab frame" or something like that. Maybe just say "Label" since other physicists may look at this code :-)